lib/callableFunction.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+var _ = require('lodash');
+var request = require('request');
+
+var _isDatabaseFunc = function(eventTrigger) {
+  return /firebase.database/.test(eventTrigger.eventType);
+};
+
+var _substituteParams = function(resource, params) {
+  var wildcards = resource.match(new RegExp('{[^/{}]*}', 'g'));
+  _.forEach(wildcards, function(wildcard) {
+    var wildcardNoBraces = wildcard.slice(1, -1); // .slice removes '{' and '}' from wildcard
+    var sub = _.get(params, wildcardNoBraces);
+    if (sub) {
+      resource = _.replace(resource, wildcard, sub);
+    } else {
+      resource = _.replace(resource, wildcard, wildcardNoBraces + '1');
+    }
+  });
+  return resource;
+};
+
+var _call = function(data, opts) {
+  opts = opts || {};
+  if (this.httpsTrigger) {
+    this.controller.call(this.name, data || {});
+  } else if (this.eventTrigger) {
+    if (_isDatabaseFunc(this.eventTrigger)) {
+      var operationType = _.last(this.eventTrigger.eventType.split('.'));
+      var dataPayload;
+      if (operationType === 'create') {
+        dataPayload = {
+          data: null,
+          delta: data
+        };
+      } else if (operationType === 'update' || operationType === 'write') {
+        dataPayload = {
+          data: data.before,
+          delta: data.after
```

Comment:
This doesn't feel right to me. First, why are we massaging the data into `before` and `after`? Is that part of a planned change to the underlying event structure? Second, `delta` is explicitly **not** the same as the data after, it is the difference between the data before and the data after. I'm worried people will end up with inaccurate local testing if they grow misconceptions based on how this functions vs. how it works live (for instance, `#changed` on delta snapshots will be busted in this impl).